airflow/sensors/base.py

```diff
@@ -114,15 +118,27 @@ def _orig_start_date(
     )
 
 
+class SkipPolicy(str, enum.Enum):
+    """Action to TODO."""
+
+    NONE = "none"
+
+    # If poke method raises an exception, sensor will be skipped.
+    SKIP_ON_ANY_ERROR = "skip_on_any_error"
+
+    SKIP_ONLY_SOFT_ERROR = "skip_only_soft_error"
+    IGNORE_ERRORS = "ignore_error"
+
+
 class BaseSensorOperator(BaseOperator, SkipMixin):
     """
     Sensor operators are derived from this class and inherit these attributes.
 
     Sensor operators keep executing at a time interval and succeed when
     a criteria is met and fail if and when they time out.
 
-    :param soft_fail: Set to true to mark the task as SKIPPED on failure.
-           Mutually exclusive with never_fail.
+    :param soft_fail: Set to true to mark the task as SKIPPED on AirflowSensorTimeout,
+           AirflowTaskTimeout, AirflowFailException.
```

Comment:
1. Mutual exclusiveness with `silent_fail` should be clarified.
2. It should be marked as deprecated.